src/GitUtils.ps1

```diff
@@ -415,6 +415,52 @@ function Get-SshAgent() {
     return 0
 }
 
+function Get-NativeSshAgent {
+    # The ssh.exe binary version must include "OpenSSH"
+    # The windows ssh-agent service must exist
+    $service = Get-Service ssh-agent
```

Comment:
`-ErrorAction Ignore`?

Reply:
Should we firewall this function to be Windows only?  If I ran this on PS Core on Linux/macOS, it would choke on `Get-Service` and `FileVersionInfo`

Reply:
Note this is targeting v0, so xplat is a nice-to-have. Easy enough to fix, regardless, if we intend posh-ssh to be xplat.

Reply:
Added `-ErrorAction Ignore` and added OS checks using PS Core's `$IsMacOs` and `$IsLinux` (undefined on PS 5). Let me know if you'd prefer a different mechanism for doing the OS check